dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -62,6 +64,124 @@ protected WorkflowResource(final WorkflowHelper workflowHelper,
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response findSchemesByContentType(@Context                         final HttpServletRequest request,
+                                                   @PathParam("contentTypeId")      final String contentTypeId) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
+        List<WorkflowScheme> schemes;
+        Map<String, Object>  resultMap;
+
+        try {
+
+            Logger.debug(this,
+                    "Getting the workflow schemes for the contentTypeId: " + contentTypeId);
+            schemes   = this.workflowHelper.findSchemesByContentType
+                    (contentTypeId, initDataObject.getUser());
+            response  =
+                    Response.ok(new ResponseEntityView(schemes)).build(); // 200
+        } catch (Exception e) {
+
+            Logger.error(this.getClass(),
+                    "Exception on findSchemesByContentType exception message: " + e.getMessage(), e);
+            response = (e.getCause() instanceof SecurityException)?
+                    ExceptionMapperUtil.createResponse(e, Response.Status.UNAUTHORIZED) :
+                    ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+
+        return response;
+    } // findSchemesByContentType.
+
+    /**
+     * Returns all schemes for the content type and include schemes non-archive . 401 if the user does not have permission.
+     * @param request  HttpServletRequest
+     * @return Response
+     */
+    @GET
+    @Path("/schemes/schemescontenttypes/{contentTypeId}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response findAllSchemesAndSchemesByContentType(@Context            final HttpServletRequest request,
+                                                   @PathParam("contentTypeId")      final String contentTypeId) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
+        List<WorkflowScheme> schemes;
+        List<WorkflowScheme> contentTypeSchemes;
+        Map<String, Object>  resultMap;
+
+        try {
+
+            Logger.debug(this,
+                    "Getting the workflow schemes for the contentTypeId: " + contentTypeId
+                            + " and including All Schemes");
+            schemes   = this.workflowHelper.findSchemes();
+            contentTypeSchemes = this.workflowHelper.findSchemesByContentType
+                    (contentTypeId, initDataObject.getUser());
+
+            response  =
+                    Response.ok(new ResponseEntityView(new SchemesAndSchemesContentTypeView
+                            (schemes, contentTypeSchemes))).build(); // 200
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)

Reply:
Not needed